lxd/db/generate/db/stmt.go

```diff
@@ -384,10 +384,14 @@ func (s *Stmt) update(buf *file.Buffer) error {
 
 	for i, field := range fields {
 		if field.IsScalar() {
-			// TODO: make this more general
-			ref := lex.Snake(field.Name)
-			updates[i] = fmt.Sprintf("%s_id = ", ref)
-			updates[i] += fmt.Sprintf("(SELECT id FROM %s WHERE name = ?)", lex.Plural(ref))
+			table, column, err := field.ScalarTableColumn()
+			if err != nil {
+				return err
+			}
+
+			// Assumes that the reference column for table "things" is of the form "thing_id"
+			updates[i] = fmt.Sprintf("%s_id = ", lex.Singular(table))
```

Comment:
This could also be updated once the `joinon` change is merged:
```go
columnName := field.Config.Get("joinon")
if columnName == "" {
  columnName = fmt.Sprintf("%s_id", lex.Singular(table))
}

updates[i] = fmt.Sprintf("%s = ", columnName)
```

Reply:
Sounds good

Reply:
Thats merged now